react/src/pages/AgentsPage.tsx

```diff
@@ -0,0 +1,118 @@
+import Flex from '../components/Flex';
+import { useWebUINavigate } from '../hooks';
+import { Agent, useAgents } from '../hooks/useAgents';
+import { Avatar, Card, List, Skeleton, Tag, theme } from 'antd';
+import { createStyles } from 'antd-style';
+import React, { Suspense } from 'react';
+
+const useStyles = createStyles(({ css, token }) => {
+  return {
+    cardList: css`
+      .and-col {
+        height: calc(100% - ${token.marginMD});
+      }
+      .ant-tag {
+        margin-inline-end: 0;
+      }
+    `,
+  };
+});
+
+const { Meta } = Card;
+
+const AgentCard = ({ agent }: { agent: Agent }) => {
+  const tags = agent.meta.tags || [];
+  return (
+    <Card
+      hoverable
+      style={{
+        height: '100%',
+      }}
+    >
+      <Flex
+        direction="column"
+        align="stretch"
+        gap="xs"
+        justify="between"
+        style={{ minHeight: '200px' }}
+      >
+        <Meta
+          title={agent.meta.title}
+          avatar={<Avatar src={agent.meta.avatar} size={150} />}
+          description={agent.meta.descriptions}
+        />
+        <Flex
+          direction="row"
+          justify="start"
+          style={{ width: '100%', flexShrink: 1 }}
+          gap={6}
+          wrap="wrap"
+        >
+          <Tag key={agent.endpoint} color={'orange-inverse'}>
+            {agent.endpoint}
+          </Tag>
+          {tags.map((tag, index) => (
+            <Tag key={index}>{tag}</Tag>
+          ))}
+        </Flex>
+      </Flex>
+    </Card>
+  );
+};
+
+const AgentCardList = ({
+  agents,
+  onClickAgent,
+}: {
+  agents: Agent[];
+  onClickAgent: (agent: Agent) => void;
+}) => {
+  const { styles } = useStyles();
+
+  return (
+    <List
+      className={styles.cardList}
+      grid={{ gutter: 16, column: 2 }}
```

Comment:
For the responsive grid,
```suggestion
      grid={{ gutter: 16, xs: 1, sm: 1, md: 2, lg: 2, xl: 3, xxl: 4 }}
```

Reply:
It is a same prop with [ModelStoreListPage](https://github.com/lablup/backend.ai-webui/blob/feature/model-player-UI-updated/react/src/pages/ModelStoreListPage.tsx#L216). We have to same layout through the pages. which one is?

Reply:
We need to implement a responsive design to prevent layout issues on small and XXL screens, as shown below:

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/XqC2uNFuj0wg8I60sMUh/cb2ccf56-6366-4720-b287-9047c3286c9b.png)

Reply:
[rd.mp4 <span class="graphite__hidden">(uploaded via Graphite)</span> <img class="graphite__hidden" src="https://app.graphite.dev/api/v1/graphite/video/thumbnail/JFysNkoR8J15CnzOUuvH/c043361d-c4a4-4ffc-90df-9b6e7ed09fa3.mp4" />](https://app.graphite.dev/media/video/JFysNkoR8J15CnzOUuvH/c043361d-c4a4-4ffc-90df-9b6e7ed09fa3.mp4)

Fixed to show 4 lines